examples/borrow_reference/sources/borrow_test.move

```diff
@@ -0,0 +1,54 @@
+module rooch_examples::borrow_test {
+    use rooch_examples::borrowd::{BorrowCapability, DataStore};
+    use rooch_examples::borrowd;
+    use moveos_std::account_storage;
+    use moveos_std::storage_context::{StorageContext};
+
+    #[test_only]
+    use std::signer;
+    #[test_only]
+    use moveos_std::storage_context;
+
+    struct Capabilities has key {
+        borrow_cap: BorrowCapability,
+    }
+
+    struct DataStoreWrapper has key {
+        data_store: DataStore,
+    }
+
+    public fun init_borrow(
+        ctx: &mut StorageContext,
+        account: &signer,
+    ) {
+        let borrow_cap = borrowd::new_borrow_cap();
+        let data_store = borrowd::new_data_store();
+        account_storage::global_move_to<Capabilities>(ctx, account, Capabilities {
+            borrow_cap,
+        });
+        account_storage::global_move_to<DataStoreWrapper>(ctx, account, DataStoreWrapper {
+            data_store,
+        });
+    }
+
+    public fun borrow(
+        ctx: &mut StorageContext,
+        addr: address,
+    ) {
+        let cap = account_storage::global_borrow_mut<Capabilities>(ctx, addr);
+        borrowd::do_immutable_borrow(ctx, &cap.borrow_cap);
+
+        //  Invalid usage of reference as function argument. Cannot transfer a mutable reference that is being borrowed
+        let data_store_warpper = account_storage::global_borrow_mut<DataStoreWrapper>(ctx, addr);
+        borrowd::do_mutable_borrow(ctx, addr, &mut data_store_warpper.data_store);
```

Comment:
mutable borrow reference error, how to avoid? @jolestar 

```
BUILDING borrow_reference
error[E07005]: invalid transfer of references
   ┌─ ./sources/borrow_test.move:43:9
   │
43 │         borrowd::do_mutable_borrow(ctx, addr, &mut data_store_warpper.data_store);
   │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   │         │                                     │
   │         │                                     It is still being mutably borrowed by this reference
   │         Invalid usage of reference as function argument. Cannot transfer a mutable reference that is being borrowed

```

Reply:
Because the `DataStoreWrapper`'s mut reference is binding to `ctx`, you need to remove the `ctx` argument from the `do_mutable_borrow` function.

Reply:
What should I do if `do_mutable_borrow` strongly depend on `ctx`?

Reply:
We must release the reference binding to `ctx`, then pass `ctx` to another function.
We need more cases to verify the `ctx` design.